operators/src/processing/expression.rs

```diff
@@ -0,0 +1,361 @@
+use crate::engine::{
+    InitializedOperator, InitializedOperatorImpl, InitializedRasterOperator, Operator,
+    QueryContext, QueryProcessor, QueryRectangle, RasterOperator, RasterQueryProcessor,
+    RasterResultDescriptor, TypedRasterQueryProcessor,
+};
+use crate::opencl::cl_program::{CLProgram, CompiledCLProgram, IterationType, RasterArgument};
+use crate::util::Result;
+use crate::{call_bi_generic_processor, call_generic_raster_processor};
+use futures::stream::BoxStream;
+use futures::StreamExt;
+use geoengine_datatypes::raster::{Pixel, Raster, Raster2D, RasterDataType, RasterTile2D};
+use geoengine_datatypes::spatial_reference::SpatialReferenceOption;
+use serde::{Deserialize, Serialize};
+use snafu::ensure;
+use std::collections::HashSet;
+use std::convert::TryFrom;
+use std::marker::PhantomData;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+pub struct ExpressionParams {
+    pub expression: String,
+    pub output_type: RasterDataType,
+}
+
+// TODO: custom type or simple string?
```

Comment:
Since we have to put a string to OpenCL, we can keep it as a string as long as it is safe.